kernel/objects/capability-spinning/objects/CapEntry.cc

```diff
@@ -125,28 +137,41 @@ namespace mythos {
     return true;
   }
 
-  optional<void> CapEntry::unlink()
+  bool CapEntry::kill(Cap expected)
   {
-    auto next = Link(_next).withoutFlags();
-    auto prev = Link(_prev).withoutFlags();
-    next->_prev.store(prev.value());
-    prev->_next.store(next.value());
-    _prev.store(Link().value());
+    CapValue expectedValue = expected.value();
+    MLOG_DETAIL(mlog::cap, this, ".kill", DVAR(expected));
+    return _cap.compare_exchange_strong(expectedValue, expected.asZombie().value());
+  }
+
+
+  optional<void> CapEntry::unlinkAndUnlockLinks()
```

Comment:
Splitting into locking and an ending operation (similar to a commit) allows us to use locked values to implement higher level operation. We can check if this is possible move locking into `unlink` after making sure we will not need that in the reworked revocation implementation.

Also, the _external_ revoke operation is not just any operation and is bound to now details about the `CapEntry`/locking anyway.

Reply:
Should we apply this acquire-commit design to the insertAfter operation as well? This might remove this one callback function via template argument in CapEntry.hh